ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/util/CrosslinkCommitteeUtil.java

```diff
@@ -14,31 +14,138 @@
+      long position = max(UnsignedLong.valueOf(index), UnsignedLong.valueOf(flip)).longValue();
+
+      Bytes sourceBytes = Bytes.concatenate(seed, roundBytes, int_to_bytes(Math.floorDiv(position, 256l), 4));
+      Bytes32 source = Hash.sha2_256(sourceBytes);
+
+      int sourceByteIndex = toIntExact(Math.floorDiv((position % 256), 8l));
+      byte byteValue = source.get(sourceByteIndex);
+      int bit = (byteValue >> (position % 8)) % 2;
+      index = (bit == 1) ? flip : index;
+    }
+    return index;
+  }
+
+  /**
+   * Computes indices of a new committee based upon the seed parameter
+   *
+   * @param indices
+   * @param seed
+   * @param index
+   * @param count
+   * @return
+   *
+   * @see <a>https://github.com/ethereum/eth2.0-specs/blob/v0.7.1/specs/core/0_beacon-chain.md#compute_committee</a>
+   */
+  public static List<Integer> compute_committee(List<Integer> indices, Bytes32 seed, int index, int count){
+    int start = Math.floorDiv(indices.size() * index, count);
+    int end = Math.floorDiv(indices.size() * (index + 1), count);
+    for(int i = start; i < end; i++)indices.set(i, get_shuffled_index(i, indices.size(), seed));
+    return indices;
   }
-  // Return the number of committees in the previous epoch of the given ``state`
-  public static int get_current_epoch_committee_count(BeaconState state) {
-    List<Validator> previous_active_validators =
-        ValidatorsUtil.get_active_validators(
-            state.getValidator_registry(), state.getCurrent_shuffling_epoch());
-    return previous_active_validators.size();
+
+  /**
+   * Convenience method for getting a crosslink committee
+   *
+   * @param state
+   * @param epoch
+   * @param shard
+   * @return
+   *
+   * @see <a>https://github.com/ethereum/eth2.0-specs/blob/v0.7.1/specs/core/0_beacon-chain.md#get_crosslink_committee</a>
+   */
+  public static List<Integer> get_crosslink_committee(BeaconState state, UnsignedLong epoch, UnsignedLong shard){
+    int index = (shard.plus(UnsignedLong.valueOf(SHARD_COUNT)).minus(get_epoch_start_shard(state, epoch)))
+            .mod(UnsignedLong.valueOf(SHARD_COUNT))
+            .intValue();
+    return compute_committee(
+            get_active_validator_indices(state, epoch),
+            generate_seed(state, epoch),
+            index,
+            get_epoch_committee_count(state, epoch).intValue()
+            );
+  }
+
+  /**
+   * Returns the index of a start shard for the provided epoch
+   *
+   * @param state
+   * @param epoch
+   * @return
+   *
+   * @see <a>https://github.com/ethereum/eth2.0-specs/blob/v0.7.1/specs/core/0_beacon-chain.md#get_epoch_start_shard</a>
+   */
+  public static UnsignedLong get_epoch_start_shard(BeaconState state, UnsignedLong epoch){
+    checkArgument(epoch.compareTo(get_current_epoch(state).plus(UnsignedLong.ONE)) <= 0, "CrosslinkCommitteeUtil.get_epoch_start_shard");
+    UnsignedLong check_epoch = get_current_epoch(state).plus(UnsignedLong.ONE);
```

Comment:
If you flip the order of this line and the one above it, you can remove the duplicate current_epoch+1 code.